addon/validators/date.js

```diff
@@ -53,9 +59,11 @@ export default Base.extend({
   },
 
   validate(value, options) {
+    options = assign({}, options);
```

Comment:
This line isn't needed here because the options object is already a new pojo created from `processOptions` method. You can also remove the `const assign` declaration from the top.